Frameworks/include/CGPathInternal.h

```diff
@@ -17,65 +17,12 @@
 #ifndef __CGPATHINTERNAL_H
 #define __CGPATHINTERNAL_H
 
+#include "CFBridgeBase.h"
 #include "CoreGraphics/CGContext.h"
 #include "CoreGraphics/CGPath.h"
-#include "CFBridgeBase.h"
-
-enum pathComponentType {
-    pathComponentRectangle,
-    pathComponentMove,
-    pathComponentLineTo,
-    pathComponentArcToPoint,
-    pathComponentQuadCurve,
-    pathComponentBezierCurve,
-    pathComponentCurve,
-    pathComponentEllipseInRect,
-    pathComponentClose,
-    pathComponentArcAngle,
-};
-
-typedef struct {
-    float x1, y1, x2, y2;
-    float radius;
-} arcToPoint;
-
-typedef struct {
-    float x, y, startAngle, endAngle;
-    float radius;
-    BOOL clockwise;
-} arcAngle;
-
-struct curveToPoint {
-    float x1, y1; // tangent from start
-    float x2, y2; // tangent to end
-    float x, y; // end pos
-};
-
-struct quadCurveToPoint {
-    float cpx, cpy;
-    float x, y;
-};
-
-struct ellipseInRect {
-    CGRect rect;
-};
-
-typedef struct {
-    pathComponentType type;
-
-    union {
-        CGRect rect;
-        CGPoint point;
-        arcToPoint atp;
-        curveToPoint ctp;
-        ellipseInRect eir;
-        quadCurveToPoint qtp;
-        arcAngle aa;
-    };
-} pathComponent;
 
 struct __CGPath : public CFBridgeBase<__CGPath> {
-    pathComponent* _components;
+    CGPathElement* _components;
```

Comment:
In light of the other comment on allocation points inline, it probably makes it easier to have CGPathElementInternal which derives from CGPathElement and adds four points.  You can then have the constructor set the pointer as well, simplifying some boilerplate.